### ISSUE TITLE
Equation numbering and referencing

### ISSUE THREAD

**Issue by xuan254** [shd101wyy/markdown-preview-enhanced#251]
Hi,

I'm trying to number my equations and reference them back in the paper when necessary. Seems the numbering does not work. May I know if there's a way to number the equations? And if possible, could I reference them back in the paragraph with something like, \ref{eq:1}?

Many thanks in advance.

**Comment by shd101wyy** [comment]
Hi @xuan254 ,

Equation numbering doesn't seem to be supported right now.  
I will have a look when I have time.

Thanks

**Comment by dpo** [comment]
@shd101wyy Do you package a full distribution of MathJax, or is it trimmed down somehow? It is odd that some standard environments, such as `\begin{equation} ... \end{equation}` and `\begin{equation*} ... \end{equation*}`, which [should be supported by default](http://docs.mathjax.org/en/latest/tex.html#environments) aren't.

As a means of debugging this, what's the way to load MathJax from the official CDN? If I change `script.src` [here](https://github.com/shd101wyy/markdown-preview-enhanced/blob/master/lib/mathjax-wrapper.coffee#L42), I get an error stating that Atom "Refused to load the script because it violates the following Content Security Policy directive".

**Comment by shd101wyy** [comment]
@dpo `begin{equation} ... \end{equation}` and `\begin{equation*}...\end{equation*}` works on my Mac.  
Make sure you write your equations inside `$...$` or `$$...$$`

**Comment by dpo** [comment]
I know about `$` and `$$`, etc. The difference is that `\begin{equation}... \end{equation}` should add an equation number. That's what I'm trying to debug.

**Comment by shd101wyy** [comment]
@xuan254  @dpo , That is because I disabled it.   
Check this line [#18](https://github.com/shd101wyy/markdown-preview-enhanced/blob/master/lib/mathjax-wrapper.coffee#L18)   
If you modify it to  

```
        MathJax.Hub.Config({
          extensions: ['tex2jax.js'],
          jax: ['input/TeX', 'output/HTML-CSS'],
          showMathMenu: false,
          messageStyle: 'none',
          tex2jax: {
            inlineMath: inline,
            displayMath: block,
            processEscapes: true,
            # processEnvironments: true,
            # preview: 'none',
            # skipTags: ["script","noscript","style","textarea"]
          },
          TeX: {
            extensions: ['AMSmath.js', 'AMSsymbols.js', 'noErrors.js', 'noUndefined.js']
            equationNumbers: { autoNumber: "AMS" }, // <== see this line 
          },
          'HTML-CSS': { availableFonts: ['TeX'] },
          skipStartupTypeset: true
        })
```

Then equation will be auto numbered.  
However, every time I modify the markdown file, the equation number will increase automatically.   
Therefore I disabled `equationNumbers: { autoNumber: "AMS" }`. (I don't know how to fix it , and I call MathJax once I update markdown file).   
It would be great if you can solve this issue.   

In conclusion, I recommend you to manually number your equation by using `\tag{number}` macro. Please check [this issue](https://github.com/shd101wyy/markdown-preview-enhanced/issues/169)  

Best regards :)

**Comment by ionm** [comment]
I have added `equationNumbers: { autoNumber: "AMS" }` in the `mathjax_config.js` file, but I could not get the equation numbering. 

Is there a reason why the `mathjax_config.js` file does not have full control over the MathJax package?

**Comment by nunoguedelha** [comment]
Hello everyone...
I've tried editing directly the main config file (menu [Atom]->[config..]) `~/.atom/config.cson` as follows:
```
"*":
  core:
    disabledPackages: [
      "markdown-preview"
      "mathjax-wrapper"
    ]
    ...
  "markdown-preview-enhanced":
    mathRenderingOption: "MathJax"
    equationNumbers: { autoNumber: "AMS" }. <== autonumbering option added here
```

And the equation numbering was activated.

Besides that, I've realized that:
- the numbering is incremented precisely by the **number of equations in the document** every time I save it, or the preview is refreshed,
- the counting is reset if you restart the application.

I wonder which command in the startup configuration initializes that counter.
